### DISCUSSION TITLE
Installation warning: invalid wheel filename

### DISCUSSION THREAD

**Discussion by masterparser** [explosion/spaCy#11894] [Help: Installation] [answered]
I am able to ultimately run spacy in a GPU-powered Jupyter Notebook provided by Kaggle. After I run `pip install -U pip setuptools wheel`, I run `pip install -U 'spacy[cuda-autodetect]'` and I see the following warning:

> WARNING: Built wheel for cupy-wheel is invalid: cupy_wheel-11.3.0-0_cupy-cuda110-py3-none-any.whl is not a valid wheel filename.
Failed to build cupy-wheel
Running setup.py install for cupy-wheel ... done
DEPRECATION: cupy-wheel was installed using the legacy 'setup.py install' method, because a wheel could not be built for it. pip 23.1 will enforce this behaviour change. A possible replacement is to fix the wheel build issue reported above.

I am not sure how to fix the wheel filename issue. As stated in the output above, cupy-wheel is eventually installed with the help of setup.py, however, this does not seem to be the preferred method.

**Comment by adrianeboyd** [comment, marked as answer]
The error messages are confusing, but this behavior is intentional for `cupy-wheel`. This package only contains a short script to auto-detect the right cupy package to install, and the script always needs to be run as a source install rather than a wheel install. `pip` makes it really hard to disable the wheel cache, so giving the cached wheels invalid names looks like it's a clunky workaround.

You can ignore these errors as long as the install works fine in the end.

If you would like to avoid the errors, you can pick the right cupy package to install directly. For most people this is currently `pip install cupy-cuda11x` or `pip install spacy[cuda11x]` for CUDA 11.2+.
